runtime/browser/devtools/xwalk_devtools_frontend.h

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
+// Copyright 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_DEVTOOLS_XWALK_DEVTOOLS_FRONTEND_H_
+#define XWALK_RUNTIME_BROWSER_DEVTOOLS_XWALK_DEVTOOLS_FRONTEND_H_
+
+#include <map>
+#include <string>
+
+#include "base/basictypes.h"
+#include "base/compiler_specific.h"
+#include "base/memory/ref_counted.h"
+#include "base/memory/scoped_ptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/values.h"
+#include "content/public/browser/devtools_agent_host.h"
+#include "content/public/browser/devtools_frontend_host.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "net/url_request/url_fetcher_delegate.h"
+#include "xwalk/runtime/browser/runtime.h"
+
+namespace base {
+class Value;
+}
+
+namespace content {
+class RenderViewHost;
+class WebContents;
+}
+
+namespace xwalk {
+
+class NativeAppWindowDesktop;
+
+class XWalkDevToolsFrontend : public content::WebContentsObserver,
+                              public content::DevToolsFrontendHost::Delegate,
+                              public content::DevToolsAgentHostClient,
+                              public net::URLFetcherDelegate,
+                              public Runtime::Observer {
+ public:
+  static XWalkDevToolsFrontend* Show(content::WebContents* inspected_contents);
+
+  void Activate();
+  void Focus();
+  void InspectElementAt(int x, int y);
+  void Close();
+
+  void DisconnectFromTarget();
+
+  NativeAppWindowDesktop* frontend_shell() const { return runtime_window_; }
```

Comment:
runtime_window_ doesn't have anything to do with XWalkDevToolsFrontend itself, it just the class is holding a reference to the actual runtime windows we're inspecting.